backend/core/src/auth/services/user.service.ts

```diff
@@ -210,21 +211,44 @@ export class UserService {
       : [await this.jurisdictionResolverService.getJurisdiction()]
     try {
       user.passwordHash = await this.passwordService.passwordToHash(password)
-      user = await this.repo.save(user)
 
-      const payload = {
-        id: user.id,
-        expiresAt: Number.parseInt(moment().add(24, "hours").format("X")),
-      }
-      user.confirmationToken = encode(payload, process.env.APP_SECRET)
-      user = await this.repo.save(user)
+      const newUser = await getManager().transaction(
+        "SERIALIZABLE",
+        async (transactionalEntityManager) => {
+          const userRepository = transactionalEntityManager.getRepository(User)
+          let newUser = await userRepository.save(user)
+
+          const payload = {
+            id: user.id,
+            expiresAt: Number.parseInt(moment().add(24, "hours").format("X")),
+          }
+          newUser.confirmationToken = encode(payload, process.env.APP_SECRET)
+          newUser = await userRepository.save(newUser)
+
+          const applicationsRepository = transactionalEntityManager.getRepository(Application)
+          const applications = await applicationsRepository
+            .createQueryBuilder("applications")
+            .leftJoinAndSelect("applications.applicant", "applicant")
+            .where("applications.user IS NULL")
+            .andWhere("applicant.emailAddress = :email", { email: newUser.email })
+            .getMany()
+
+          for (const application of applications) {
+            application.user = newUser
+          }
+
+          await applicationsRepository.save(applications)
+
+          return newUser
+        }
+      )
```

Comment:
@pbn4 ,
I think this should be split apart some, especially since `invite` calls `createUser` and that doesn't need to care about applications. Also, I didn't notice this before, but do we need to call `save` multiple times, or can we can we get everything we need and then create the user? In the case of `invite`, it could send all what it needs to an abstracted saveUser function, so we only save the user once, no?